zkevm-circuits/src/sig_circuit.rs

```diff
@@ -235,8 +253,21 @@ impl<F: Field> SubCircuit<F> for SigCircuit<F> {
     // Since sig circuit / halo2-lib use veticle cell assignment,
     // so the returned pair is consisted of same values
     fn min_num_rows_block(block: &crate::witness::Block<F>) -> (usize, usize) {
-        let row_num = Self::min_num_rows(block.circuits_params.max_txs);
-        (row_num, row_num)
+        let row_num = Self::min_num_rows();
+        
+        let tx_count = block.txs.len();
+        
+        // TODO: Take the tx limit by block? 
+        let max_tx_count = block.circuits_params.max_txs;
```

Comment:
this is correct.  Sig circuit allocates `calc_required_advices(MAX_NUM_SIG)` advice columns for halo2-libs, which may not be COLUMN_NUM_LIMIT, usually less. `COLUMN_NUM_LIMIT` is a loose limit, usually irrelevant for runtime circuit.   yes the codes here is a bit confusing eg we implicitly require block.circuits_params.max_txs <= MAX_NUM_SIG but that is not documented...